### PR TITLE
Make cspell checks case sensitive. Fixes #5198.

### LINKED ISSUE #5198: Add case checking to linter / spell checking rules
Can we get the linter to check casing?

_Originally posted by @RickWinter in https://github.com/Azure/azure-sdk-for-cpp/pull/5185#pullrequestreview-1750850783_

Maybe we could tell the spell checker to check casing? Specifically, typos like "THis".

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -4,7 +4,20 @@
...
     "powershell",
-    "cpp"
+    "cpp",
+    "markdown"
+  ],
+  "languageSettings": [
+    {
+      "languageId": "cpp",
+      "allowCompoundWords": true,
+      "caseSensitive": true
```
[the remark this answers is not in the document]
Comment (LarryOsterman) [reply .vscode/cspell.json:14]: FWIW, I ran the test locally and it finds a LOT of errors. Dozens and dozens of errors.

[the remark this answers is not in the document]
Comment (LarryOsterman) [reply .vscode/cspell.json:14]: I've fixed the errors in the "template" package and a couple of the most egregious errors in core, we have 89 errors in core left (several of which are very real issues). 

We can fix these over time.

**Comment by ahsonkhan** [comment]
> We can fix these over time.

How can we fix them over time? Won't they fail the build and block CI?

**Comment by LarryOsterman** [comment]
> > We can fix these over time.
> 
> How can we fix them over time? Won't they fail the build and block CI? Or are we treating spelling errors as warnings only?

The cspell check only looks at files which have been modified. So they can be fixed over time.

**Pull request merged**